tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -1466,22 +1475,44 @@ SpirvVariable *DeclResultIdMapper::createPushConstant(const VarDecl *decl) {
 SpirvVariable *
 DeclResultIdMapper::createShaderRecordBuffer(const VarDecl *decl,
                                              ContextUsageKind kind) {
+  const QualType type = decl->getType();
   const auto *recordType =
-      hlsl::GetHLSLResourceResultType(decl->getType())->getAs<RecordType>();
+      hlsl::GetHLSLResourceResultType(type)->getAs<RecordType>();
   assert(recordType);
 
   assert(kind == ContextUsageKind::ShaderRecordBufferEXT ||
          kind == ContextUsageKind::ShaderRecordBufferNV);
 
-  const auto typeName = kind == ContextUsageKind::ShaderRecordBufferEXT
-                            ? "type.ShaderRecordBufferEXT."
-                            : "type.ShaderRecordBufferNV.";
+  SpirvVariable *var;
+  if (isConstantBuffer(type)) {
+    // Constant buffers already have Block decoration. The variable will need
+    // the appropriate storage class.
+
+    const auto sc = kind == ContextUsageKind::ShaderRecordBufferNV
+                        ? spv::StorageClass::ShaderRecordBufferNV
+                        : spv::StorageClass::ShaderRecordBufferKHR;
```

Comment:
I don't see any test checking we do add the required capabilities/extensions. Shall this be added?
I do believe changing EXT into KHR is a good change, but maybe we should now check it.

Reply:
It looks like there are tests that the ray tracing capability and extension are added for the `RaytracingAccelerationStructure` type, but not for the `vk::shader_record_ext` attribute. I'll add them into the new tests.